### ISSUE TITLE
Error while Ossec 2.9.0 installation on CentOS 7

### ISSUE THREAD

**Issue by adutta14** [ossec/ossec-hids#1498]
Hello,

I was trying to install ossec agent on CentOS. Initially, the installation was successful but when I ran the below command to auto register it to the manager, I got "missing OpenSSL Support".

/var/ossec/bin/agent-auth -m <server-ip> -p 1515

To fix this, I followed the solution from case "https://github.com/ossec/ossec-hids/issues/194" and make sure that openssl folder exists in that location. Now, when I try to reinstall ossec agent, I get a new error now:-

 CC ossec-agentd
/bin/ld: cannot find -lssl
/bin/ld: cannot find -lcrypto
collect2: error: ld returned 1 exit status
make: *** [ossec-agentd] Error 1

 Error 0x5.
 Building error. Unable to finish the installation.
 
Please help.

Thanks,
Abhishek Dutta

**Comment by ddpbsd** [comment]
Did you install the openssl devel package?

**Comment by adutta14** [comment]
Yes, its installed. Anything specific you want me to check related to OpenssL ?

**Comment by ddpbsd** [comment]
`ldconfig -p | grep ssl` should help you figure out where that's installed. I guess you might have to make sure that location is being looked at. Using the `V=1` option might give you more clues since it shows the compilation instead of the pretty terminal stuff. I think that's in 2.9.0, but it's been so long since I've looked at that version I could be mistaken.

**Comment by ddpbsd** [comment]
Sorry. Now you know where libssl and libcrypto are installed.
Re-run the installer or make, but add `V=1` to it.
`env V=1 ./install.sh`
or
`make TARGET=agent V=1`
That will give you more information on the actual compilation process.
Instead of just seeing `CC ossec-agentd` you could see something like:
```
cc  -O2 -DMAX_AGENTS=2048 -DOSSECHIDS -DDEFAULTDIR=\"/var/ossec\" -DUSER=\"ossec\" -DREMUSER=\"ossecr\" -DGROUPGLOBAL=\"ossec\" -DMAILUSER=\"ossecm\" -DLinux -DINOTIFY_ENABLED -DLIBOPENSSL_ENABLED -DCLIENT -Wall -Wextra -I./ -I./headers/ -I./external/zlib-1.2.11/ client-agent/sendmsg.o client-agent/config.o client-agent/intcheck_op.o client-agent/agentd.o client-agent/event-forward.o client-agent/receiver-win.o client-agent/main.o client-agent/receiver.o client-agent/start_agent.o client-agent/notify.o os_crypto.a config.a shared.a os_net.a os_regex.a os_xml.a os_zlib.a external/zlib-1.2.11/libz.a  -lm -lpthread -lssl -lcrypto -o ossec-agentd
```
Of course, this doesn't mention anything about the installed location of the ssl stuff. So maybe that's the wrong path here.

```
[root@va-is-ossec src]# ldconfig -p | grep ssl
        libxmlsec1-openssl.so.1 (libc6,x86-64) => /lib64/libxmlsec1-openssl.so.1
        libxmlsec1-openssl.so (libc6,x86-64) => /lib64/libxmlsec1-openssl.so
        libssl3.so (libc6,x86-64) => /lib64/libssl3.so
        libssl.so.10 (libc6,x86-64) => /lib64/libssl.so.10
        libevent_openssl-2.0.so.5 (libc6,x86-64) => /lib64/libevent_openssl-2.0.so.5
[root@va-is-ossec src]# rpm -qa | grep openssl
openssl-libs-1.0.2k-12.el7.x86_64
xmlsec1-openssl-1.2.20-7.el7_4.x86_64
openssl-1.0.2k-12.el7.x86_64
openssl-devel-1.0.2k-12.el7.x86_64
```

**Comment by adutta14** [comment]
Yes you are right. After running the given command, I see below:-

cc  -O2 -DMAX_AGENTS=2048 -DOSSECHIDS -DDEFAULTDIR=\"/var/ossec\" -DUSER=\"ossec\" -DREMUSER=\"ossecr\" -DGROUPGLOBAL=\"ossec\" -DMAILUSER=\"ossecm\" -DLinux -DINOTIFY_ENABLED -DPICVIZ_OUTPUT_ENABLED -DLIBOPENSSL_ENABLED -DCLIENT -Wall -Wextra -I./ -I./headers/ -I./external/zlib-1.2.8/ client-agent/sendmsg.o client-agent/config.o client-agent/intcheck_op.o client-agent/agentd.o client-agent/event-forward.o client-agent/receiver-win.o client-agent/main.o client-agent/receiver.o client-agent/start_agent.o client-agent/notify.o os_crypto.a config.a shared.a os_net.a os_regex.a os_xml.a os_zlib.a external/zlib-1.2.8/libz.a  -lm -lpthread -lssl -lcrypto -o ossec-agentd
/bin/ld: cannot find -lssl
/bin/ld: cannot find -lcrypto
collect2: error: ld returned 1 exit status
make: *** [ossec-agentd] Error 1

 Error 0x5.
 Building error. Unable to finish the installation.

It doesn't say anything about where exactly is it looking for the files. So, how to proceed further from here, please ?

**Comment by ddpbsd** [comment]
I'm not sure. You could try setting the LD_LIBRARY_PATH
`env LD_LIBRARY_PATH=/lib64 V=1 ./install.sh`
There's either something odd with your system, or something's been fixed since 2.9.0 came out a year and a half ago.

**Comment by adutta14** [comment]
Well, I tried the command but still no luck. One thing which I would like to highlight again is if I remove that "**openssl**" folder from the location :-  "**/usr/include**" ,  the error goes away and the installation is successful.

**Comment by ddpbsd** [comment]
Well that has what looks to be a bunch of important stuff in it. But if that works, yay!

**Comment by adutta14** [comment]
No, but if that folder is missing from the location then I get the below error while registering the agent to the manager.

sudo /var/ossec/bin/agent-auth -m <server_IP> -p 1515 -k /var/ossec/etc/client.key
ERROR: Not compiled. Missing OpenSSL support.

**Comment by ddpbsd** [comment]
Well yeah, you removed openssl from the system (kinda sorta).

**Comment by adutta14** [comment]
I have only removed it from that location. Its still there installed in my system.

whereis openssl
openssl: /usr/bin/openssl /usr/lib64/openssl /usr/include/openssl.tar.gz /usr/share/man/man1/openssl.1ssl.gz

Is there a way I can make the agent-auth program to look at these locations for openssl instead of "usr/include/openssl"?

**Comment by ddpbsd** [comment]
You could try mucking around in the Makefile.
Maybe try changing `-lssl` to `-l:libssl.so.10` or something

**Comment by adutta14** [comment]
So, you mean should I try to change the Makefile and then reinstall the agent again or only run "/var/ossec/bin/agent-auth -m <server_IP> -p 1515 -k /var/ossec/etc/client.key" command?

**Comment by ddpbsd** [comment]
Ok, I'm confused. I thought the agent-auth command wasn't working because it isn't picking up the openssl-devel installation?

**Comment by adutta14** [comment]
Well, even making the changes in the Makefile doesn't help.
